bot.py

```diff
@@ -112,7 +112,28 @@ async def list(ctx):
     )
 
 
-@bot.command()
+# Support inv [add|remove]
+@bot.group()
+async def inv(ctx):
+    if ctx.invoked_subcommand is None:
+        author = ctx.message.author.id
+        name = ctx.message.author.name
+        one_line = "\n".join(tracker.get_inv(author))
+        await ctx.message.channel.send(
+            embed = Embed().from_dict(
+                {
+                    "fields": [
+                        {
+                            "name": f"__*{name}'s Inventory:*__",
+                            "value": one_line,
```

Comment:
No spaces around `=` during a function call. I think these names are more descriptive:

- `author_id`: `author` suggests to me that I probably have a `User` object where in fact it is only the ID.
- `author_name`
- `one_per_line`

I would be happy collapsing most of it since we only use each exactly once and we don't lose any clarity. However, empty inventories will fail completely as `value` becomes an empty string and that's not allowed (`HTTPException: 400 Bad Request`) so we have to guard against that at least when we try to report an empty inventory back to the author.

```suggestion
        one_per_line = "\n".join(tracker.get_inv(ctx.message.author.id))
        if len(one_per_line) == 0:
            one_per_line = "<< Empty >>"
        await ctx.message.channel.send(
            embed = Embed().from_dict(
                {
                    "fields": [
                        {
                            "name": f"__*{ctx.message.author.name}'s Inventory:*__",
                            "value": one_per_line,
```

Reply:
Suggested changes made. Thanks!